physics/module_mp_thompson.F90

```diff
@@ -92,142 +92,142 @@ MODULE module_mp_thompson
 
 !..Capacitance of sphere and plates/aggregates: D**3, D**2
-      REAL, PARAMETER, PRIVATE:: C_cube = 0.5
-      REAL, PARAMETER, PRIVATE:: C_sqrd = 0.15
+      REAL(kind_phys), PARAMETER, PRIVATE:: C_cube = 0.5
+      REAL(kind_phys), PARAMETER, PRIVATE:: C_sqrd = 0.15
 
 !..Collection efficiencies.  Rain/snow/graupel collection of cloud
 !.. droplets use variables (Ef_rw, Ef_sw, Ef_gw respectively) and
 !.. get computed elsewhere because they are dependent on stokes
 !.. number.
-      REAL, PARAMETER, PRIVATE:: Ef_si = 0.05
-      REAL, PARAMETER, PRIVATE:: Ef_rs = 0.95
-      REAL, PARAMETER, PRIVATE:: Ef_rg = 0.75
-      REAL, PARAMETER, PRIVATE:: Ef_ri = 0.95
+      REAL(kind_phys), PARAMETER, PRIVATE:: Ef_si = 0.05
+      REAL(kind_phys), PARAMETER, PRIVATE:: Ef_rs = 0.95
+      REAL(kind_phys), PARAMETER, PRIVATE:: Ef_rg = 0.75
+      REAL(kind_phys), PARAMETER, PRIVATE:: Ef_ri = 0.95
 
 !..Minimum microphys values
 !.. R1 value, 1.E-12, cannot be set lower because of numerical
 !.. problems with Paul Field's moments and should not be set larger
 !.. because of truncation problems in snow/ice growth.
-      REAL, PARAMETER, PRIVATE:: R1 = 1.E-12
-      REAL, PARAMETER, PRIVATE:: R2 = 1.E-6
-      REAL, PARAMETER :: eps = 1.E-15
+      REAL(kind_phys), PARAMETER, PRIVATE:: R1 = 1.E-12
+      REAL(kind_phys), PARAMETER, PRIVATE:: R2 = 1.E-6
+      REAL(kind_phys), PARAMETER :: eps = 1.E-15
 
 !..Constants in Cooper curve relation for cloud ice number.
-      REAL, PARAMETER, PRIVATE:: TNO = 5.0
-      REAL, PARAMETER, PRIVATE:: ATO = 0.304
+      REAL(kind_phys), PARAMETER, PRIVATE:: TNO = 5.0
+      REAL(kind_phys), PARAMETER, PRIVATE:: ATO = 0.304
 
 !..Rho_not used in fallspeed relations (rho_not/rho)**.5 adjustment.
-      REAL, PARAMETER, PRIVATE:: rho_not = 101325.0/(287.05*298.0)
+      REAL(kind_phys), PARAMETER, PRIVATE:: rho_not = 101325.0/(287.05*298.0)
 
 !..Schmidt number
-      REAL, PARAMETER, PRIVATE:: Sc = 0.632
-      REAL, PRIVATE:: Sc3
+      REAL(kind_phys), PARAMETER, PRIVATE:: Sc = 0.632
+      REAL(kind_phys), PRIVATE:: Sc3
 
 !..Homogeneous freezing temperature
-      REAL, PARAMETER, PRIVATE:: HGFR = 235.16
+      REAL(kind_phys), PARAMETER, PRIVATE:: HGFR = 235.16
 
 !..Water vapor and air gas constants at constant pressure
-      REAL, PARAMETER, PRIVATE:: Rv = 461.5
-      REAL, PARAMETER, PRIVATE:: oRv = 1./Rv
-      REAL, PARAMETER, PRIVATE:: R = 287.04
-      REAL, PARAMETER, PRIVATE:: Cp = 1004.0
-      REAL, PARAMETER, PRIVATE:: R_uni = 8.314                           !< J (mol K)-1
+      REAL(kind_phys), PARAMETER, PRIVATE:: Rv = 461.5
+      REAL(kind_phys), PARAMETER, PRIVATE:: oRv = 1./Rv
+      REAL(kind_phys), PARAMETER, PRIVATE:: R = 287.04
+      REAL(kind_phys), PARAMETER, PRIVATE:: Cp = 1004.0
+      REAL(kind_phys), PARAMETER, PRIVATE:: R_uni = 8.314                           !< J (mol K)-1
 
       DOUBLE PRECISION, PARAMETER, PRIVATE:: k_b = 1.38065E-23           !< Boltzmann constant [J/K]
```

Comment:
Ideally the "double precision" should be "real(kind_dbl_phys)", from machine.F.

Reply:
I cannot determine the precision of those variables or parameters one by one. It would be better that the scheme developers explicitly cast the type when it is needed.